pkg/cmd/roachtest/tests/kv.go

```diff
@@ -896,3 +897,130 @@ func registerKVRangeLookups(r registry.Registry) {
 		})
 	}
 }
+
+// measureQPS will measure the approx QPS at the time this command is run. The
+// duration is how long of an interval to wait while measuring. Setting too
+// short of an interval can mean inaccuracy in results. Setting too long of an
+// interval may mean the impact is blurred out.
+func measureQPS(ctx context.Context, t test.Test, db *gosql.DB, duration time.Duration) float64 {
+	numInserts := func() float64 {
+		var v float64
+		if err := db.QueryRowContext(
+			ctx, `SELECT value FROM crdb_internal.node_metrics WHERE name = 'sql.insert.count'`,
+		).Scan(&v); err != nil {
+			t.Fatal(err)
+		}
+		return v
+	}
+
+	before := numInserts()
+	time.Sleep(duration)
+	after := numInserts()
+	return (after - before) / duration.Seconds()
+}
+
+// registerKVRestartImpact measures the impact of stopping and then restarting a
+// node during a write-heavy workload. Specifically the Raft log on the node
+// falls behind when the node is down and when it comes back up it goes into IO
+// Overload as it attempts to recover. Note that this test stops the replicate
+// queue during the test to help isolate the impact of Raft backlog vs snapshot
+// transfers.
+func registerKVRestartImpact(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Skip:    "#95159",
+		Name:    "kv/restart/nodes=12",
+		Owner:   registry.OwnerKV,
+		Cluster: r.MakeClusterSpec(13, spec.CPU(8)),
+		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+			nodes := c.Spec().NodeCount - 1
+			workloadNode := c.Spec().NodeCount
+			c.Put(ctx, t.Cockroach(), "./cockroach", c.All())
+			c.Start(ctx, t.L(), option.DefaultStartOpts(), install.MakeClusterSettings(), c.Range(1, nodes))
+
+			// The duration of the outage.
+			duration, err := time.ParseDuration(ifLocal(c, "20s", "10m"))
+			assert.NoError(t, err)
+			// Set the duration of the entire test to be 3x the outage duration.
+			testDuration := 3 * duration
+
+			db := c.Conn(ctx, t.L(), 1)
+			defer db.Close()
+
+			t.Status("initializing kv dataset <", 3*time.Minute)
```

Comment:
[tiny nit] Instead of all the string concatenation here and below in t.Status, use t.Status(fmt.Sprintf("...", ...)).